src/test/java/seedu/jarvis/model/finance/FinanceTrackerTest.java

```diff
@@ -0,0 +1,121 @@
+    }
+
+    @Test
+    public void deletePayment_normalInput_deletedCorrectly() {
+        Purchase deletedPurchase = financeTracker.deleteSinglePayment(2);
+        assertEquals(new PurchaseStub().getDescription(), deletedPurchase.getDescription());
+        assertEquals(new PurchaseStub().getMoneySpent(), deletedPurchase.getMoneySpent());
+        assertEquals(2, financeTracker.getTotalPurchases());
+    }
+
+    @Test
+    public void deletePayment_indexNonexistent_throwsError() {
+        assertThrows(IndexOutOfBoundsException.class, () -> financeTracker.deleteSinglePayment(4));
+        assertEquals(3, financeTracker.getTotalPurchases());
+    }
+
+    @Test
+    public void addInstallment_normalInput_addedCorrectly() {
+        financeTracker.addInstallment(new InstallmentStub());
+        Installment addedInstalment = financeTracker.getInstallment(4);
+        assertEquals(new InstallmentStub().getDescription(), addedInstalment.getDescription());
+        assertEquals(new InstallmentStub().getMoneySpentOnInstallment(), addedInstalment.getMoneySpentOnInstallment());
+        assertEquals(4, financeTracker.getTotalInstallments());
+    }
+
+    @Test
+    public void deleteInstallment_normalInput_deletedCorrectly() {
+        Installment deletedInstallment = financeTracker.deleteInstallment(2);
+        assertEquals(new InstallmentStub().getDescription(), deletedInstallment.getDescription());
+        assertEquals(new InstallmentStub().getMoneySpentOnInstallment(),
+                deletedInstallment.getMoneySpentOnInstallment());
+        assertEquals(2, financeTracker.getTotalInstallments());
+    }
+
+    @Test
+    public void deleteInstallment_indexNonexistent_throwsError() {
+        assertThrows(IndexOutOfBoundsException.class, () -> financeTracker.deleteInstallment(4));
+        assertEquals(3, financeTracker.getTotalInstallments());
+    }
+
+    @Test
+    public void editInstallment_normalInputs_editedCorrectly() {
+        financeTracker.editInstallment(1,
+                "Student price Spotify subscription", 7.50);
+        assertEquals("Student price Spotify subscription",
+                financeTracker.getInstallment(1).getDescription());
+        assertEquals(7.50, financeTracker.getInstallment(1).getMoneySpentOnInstallment());
+    }
+
+    @Test
+    public void editInstallment_indexNonexistent_throwsError() {
+        assertThrows(IndexOutOfBoundsException.class, (
+
+        ) -> financeTracker.editInstallment(5, "Spotify", 9.50));
+    }
+
+    @Test
+    public void editInstallment_emptyDescription_throwsError() {
+        assertThrows(NullPointerException.class, (
+
+            ) -> financeTracker.editInstallment(3, null, 9.50));
+    }
+
+    @Test
+    public void setMonthlyLimit_normalInput_updatedCorrectly() {
+        financeTracker.setMonthlyLimit(500.0);
+        assertEquals(500.0, financeTracker.getMonthlyLimit());
+    }
+
+}
+
+class PurchaseStub2 extends Purchase {
```

Comment:
Lets make all stub classes private, so there is no conflict in stub class names or careless mistakes in using the wrong stub in a different test.

Reply:
alright, thanks for the tip!